erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -182,6 +182,7 @@ def update_prospect(self):
 				prospect.append("opportunities", opportunity_values)
 				prospect.flags.ignore_permissions = True
 				prospect.flags.ignore_mandatory = True
+				prospect.flags.ignore_links = True
```

Comment:
Why lose data integrity?

Reply:
When creating opportunity from prospect, the opportunity is not created yet and the link check in prospect fails unfortunately.

Reply:
I would fail with the error described in issue #40566